packages/flutter_tools/test/general.shard/resident_runner_test.dart

```diff
@@ -336,21 +337,32 @@ void main() {
 
     // supports service protocol
     expect(residentRunner.supportsServiceProtocol, true);
-    expect(bufferLogger.statusText, contains('"w"'));
-    expect(bufferLogger.statusText, contains('"t"'));
-    expect(bufferLogger.statusText, contains('"P"'));
-    expect(bufferLogger.statusText, contains('"a"'));
     // isRunningDebug
     expect(residentRunner.isRunningDebug, true);
-    expect(bufferLogger.statusText, contains('"L"'));
-    expect(bufferLogger.statusText, contains('"S"'));
-    expect(bufferLogger.statusText, contains('"U"'));
-    expect(bufferLogger.statusText, contains('"i"'));
-    expect(bufferLogger.statusText, contains('"p"'));
-    expect(bufferLogger.statusText, contains('"o"'));
-    expect(bufferLogger.statusText, contains('"z"'));
-    // screenshot
-    expect(bufferLogger.statusText, contains('"s"'));
+    // commands
+    expect(bufferLogger.statusText, equals(
+        <dynamic>[
+          'Flutter run key commands.',
+          CommandHelp.r(),
+          CommandHelp.R(),
+          CommandHelp.h(),
+          CommandHelp.q(),
+          CommandHelp.s(),
+          CommandHelp.w(),
+          CommandHelp.t(),
+          CommandHelp.L(),
+          CommandHelp.S(),
+          CommandHelp.U(),
+          CommandHelp.i(),
+          CommandHelp.p(),
+          CommandHelp.o(),
+          CommandHelp.z(),
+          CommandHelp.P(),
+          CommandHelp.a(),
+          'An Observatory debugger and profiler on null is available at:\nfoo://bar',
+          ''
+        ].join('\n')
+    ));
   }));
```

Comment:
we should have some tests that test CommandHelp.toString()'s logic for various combinations of inputs and terminal widths and so on.

Reply:
I agree the reason I left those tests out was to get your feedback on the refactoring first before I invest too much in it.

Reply:
This is great. My feedback is positive. :-)